miri/validation.rs

```diff
@@ -558,6 +560,7 @@ impl<'a, 'tcx> EvalContextExt<'tcx> for EvalContext<'a, 'tcx, super::Evaluator<'
             TyAdt(adt, _) if adt.is_box() => true,
             TySlice(_) | TyAdt(_, _) | TyTuple(..) | TyClosure(..) | TyArray(..) |
             TyDynamic(..) | TyGenerator(..) | TyForeign(_) => false,
+            TyGeneratorWitness(..) => bug!("I'm not sure what to return here"),
```

Comment:
What should I return here?

Reply:
I don't think this is reachable, so just throw an `unreachable!()` invocation there for now

Reply:
Sounds reasonable.